kafka-rest/src/main/java/io/confluent/kafkarest/response/StreamingResponse.java

```diff
@@ -145,7 +168,10 @@ private ResultOrError handleNext(T result, @Nullable Throwable error) {
     private InputStreamingResponse(
         MappingIterator<T> mappingIteratorInput, ChunkedOutputFactory chunkedOutputFactory) {
       super(chunkedOutputFactory);
-      this.mappingIteratorInput = requireNonNull(mappingIteratorInput);
+      if (mappingIteratorInput == null) {
+        throw new BadRequestException("Null input provided. Data is required.");
+      }
```

Comment:
Is this possible? That's surprising. I think it would be better to check it in the `ProduceAction` before calling the `StreamingResponse`. The correct exception for `null` is NPE, like it was before.

Reply:
Yup, easy to hit:

 curl -X POST  -H "Content-Type: application/json"   "http://localhost:8082/v3/clusters/lkc-xrn7x1/topics/bob/records"  
{"error_code":400,"message":"Bad Request: Null input provided. Data is required."}%   

I don't think it should throw a NPE, that would cause a 500 error.  We just need to tell the user that they need to send us some data.

Reply:
(I moved this to the produce action though, as suggested)